src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -174,34 +200,97 @@ CreateSpaceProcessor::pickHosts(PartitionID partId,
+CreateSpaceProcessor::pickHostsWithZone(const std::vector<std::string>& zones,
+                                        std::unordered_map<HostAddr, int32_t>& loading) {
+    std::vector<HostAddr> pickedHosts;
+    for (auto& zoneName : zones) {
         auto zoneKey = MetaServiceUtils::zoneKey(zoneName);
         auto zoneValueRet = doGet(std::move(zoneKey));
+        auto hosts = MetaServiceUtils::parseZoneHosts(std::move(zoneValueRet).value());
+
+        HostAddr picked;
+        int32_t size = INT_MAX;
+        for (auto& host : hosts) {
+            auto hostIter = loading.find(host);
+            if (hostIter == loading.end()) {
+                LOG(ERROR) << "Host " << host << " not found";
+                handleErrorCode(cpp2::ErrorCode::E_NO_HOSTS);
+                onFinished();
+                return Status::Error("Host not found");
+            }
+
+            if (size > hostIter->second) {
+                picked = host;
+                size = hostIter->second;
+            }
+        }
+
+        loading[picked] += 1;
+        pickedHosts.emplace_back(toThriftHost(std::move(picked)));
+    }
+    return pickedHosts;
+}
+
+StatusOr<std::vector<std::string>>
+CreateSpaceProcessor::pickLightLoadZones(const std::vector<std::string>& zones,
+                                         int32_t replicaFactor,
+                                         std::unordered_map<HostAddr, int32_t>& loading) {
+    std::unordered_map<std::string, int32_t> zoneLoading;
+    for (auto& zone : zones) {
+        auto zoneKey = MetaServiceUtils::zoneKey(zone);
+        auto zoneValueRet = doGet(std::move(zoneKey));
         if (!zoneValueRet.ok()) {
-            LOG(ERROR) << "Get zone " << zoneName << " failed";
-            handleErrorCode(cpp2::ErrorCode::E_STORE_FAILURE);
-            onFinished();
-            return pickedHosts;
+            LOG(ERROR) << "Get zone " << zone << " failed";
+            return Status::Error("Get Zone Failed");
         }
+
         auto hosts = MetaServiceUtils::parseZoneHosts(std::move(zoneValueRet).value());
-        pickedHosts.emplace_back(toThriftHost(hosts[startIndex++ % hosts.size()]));
+        for (auto& host : hosts) {
+            auto hostIter = loading.find(host);
+            if (hostIter == loading.end()) {
+                LOG(ERROR) << "Host " << host << " not found";
+                return Status::Error("Host not found");
+            }
+            zoneLoading[zone] += hostIter->second;
+        }
     }
-    return pickedHosts;
+
+    std::multimap<int32_t, std::string> sortedMap;
+    for (const auto &pair : zoneLoading) {
+        sortedMap.insert(std::make_pair(pair.second, pair.first));
+    }
+    std::vector<std::string> pickedZones;
+    for (const auto &pair : sortedMap) {
+        pickedZones.emplace_back(pair.second);
+        int32_t size = pickedZones.size();
+        if (size == replicaFactor) {
+            break;
+        }
+    }
+    return pickedZones;
```

Comment:
what purpose of `std::multimap<int32_t, std::string> sortedMap;` ?

Reply:
sort by value